package/tests/test_server.py

```diff
@@ -338,22 +261,151 @@ def test_no_load_file(cli_runner):
+    assert (
+        data["dataset_type"]
+        == f"{PickleDataSet.__module__}.{PickleDataSet.__qualname__}"
+    )
+
+
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_node_metadata_endpoint_data_output(cli_runner, client, tmp_path):
+    """Test `/api/nodes/data_id` endpoint is functional and returns a valid empty JSON."""
+    cli_runner.invoke(server.commands, ["viz", "--port", "8000"])
+    # 'bob_out' is not stored in DummyDataCatalog
+    response = client.get(f"/api/nodes/{_hash('bob_out')}")
+    assert response.status_code == 200
+    data = json.loads(response.data.decode())
+    assert not data
+
+
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_node_metadata_endpoint_data_kedro15(cli_runner, client, tmp_path, mocker):
+    """Test `/api/nodes/data_id` endpoint is functional and returns a valid JSON
+    with Kedro 0.15.*.
+    """
+    mocker.patch("kedro_viz.server.KEDRO_VERSION", VersionInfo.parse("0.15.0"))
+    cli_runner.invoke(server.commands, ["viz", "--port", "8000"])
+    response = client.get(f"/api/nodes/{_hash('bob_in')}")
+    assert response.status_code == 200
+    data = json.loads(response.data.decode())
+
+    assert data["dataset_location"] == str(tmp_path)
+    assert (
+        data["dataset_type"]
+        == f"{PickleDataSet.__module__}.{PickleDataSet.__qualname__}"
+    )
+
+
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_node_metadata_endpoint_data_kedro14(cli_runner, client, tmp_path, mocker):
+    """Test `/api/nodes/data_id` endpoint is functional and returns a valid JSON
+    with Kedro 0.14.*.
+    """
+    mocker.patch("kedro_viz.server.KEDRO_VERSION", VersionInfo.parse("0.14.0"))
+    cli_runner.invoke(server.commands, ["viz", "--port", "8000"])
+    response = client.get(f"/api/nodes/{_hash('bob_in')}")
+    assert response.status_code == 200
+    data = json.loads(response.data.decode())
+    assert data["dataset_location"] == str(tmp_path)
+    assert (
+        data["dataset_type"]
+        == f"{PickleDataSet.__module__}.{PickleDataSet.__qualname__}"
+    )
+
+
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_node_metadata_endpoint_params(cli_runner, client):
+    """Test `/api/nodes/param_id` endpoint is functional and returns an empty JSON."""
+    cli_runner.invoke(server.commands, ["viz", "--port", "8000"])
+    param_id = "f1f1425b"
+    response = client.get(f"/api/nodes/{param_id}")
+    assert response.status_code == 200
+    data = json.loads(response.data.decode())
+    assert not data
+
+
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_node_metadata_endpoint_invalid(cli_runner, client):
+    """Test `/api/nodes/invalid_id` endpoint returns an empty JSON."""
+    cli_runner.invoke(server.commands, ["viz", "--port", "8000"])
+    param_id = "invalid"
+    response = client.get(f"/api/nodes/{param_id}")
+    assert response.status_code == 404
```

Comment:
You should also check the body of the response, I suspect it might not be formatted as a JSON